crypto/asn1/a_d2i_fp.c

```diff
@@ -137,7 +137,7 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
                 ERR_raise(ERR_LIB_ASN1, ERR_R_BUF_LIB);
                 goto err;
             }
-            i = BIO_read(in, &(b->data[len]), want);
+            i = BIO_read(in, &(b->data[len]), (int)want);
```

Comment:
want = slen which is a long (could that be truncated?)

Reply:
Not in any realistic scenario.

Reply:
It is also actually IMO good that we won't read more than 2^31 bytes when parsing ASN.1. This PR is definitely not aiming to "move to size_t" support in general.